src/languages/smtlib/lexSmtlib.mll

```diff
@@ -42,35 +42,53 @@ rule token = parse
   | [' ' '\t' '\r']+    { token lexbuf }
   | '\n'                { Lexing.new_line lexbuf; token lexbuf }
   | ';' (_ # '\n')*     { token lexbuf }
+
   | '_'                 { UNDERSCORE }
+  | '!'                 { ATTRIBUTE }
   | "as"                { AS }
   | "let"               { LET }
-  | "forall"            { FORALL }
   | "exists"            { EXISTS }
-  | '!'                 { ATTRIBUTE }
-  | "par"               { raise Error }
+  | "forall"            { FORALL }
+  | "match"             { raise Error }
+  | "par"               { PAR }
+
   | "NUMERAL"           { raise Error }
   | "DECIMAL"           { raise Error }
   | "STRING"            { raise Error }
-  | "set-logic"         { SET_LOGIC }
-  | "set-option"        { SET_OPTION }
-  | "set-info"          { SET_INFO }
-  | "declare-sort"      { DECLARE_SORT }
-  | "define-sort"       { DEFINE_SORT }
-  | "declare-fun"       { DECLARE_FUN }
-  | "define-fun"        { DEFINE_FUN }
-  | "push"              { PUSH }
-  | "pop"               { POP }
+
+  | "not"               { NOT }
```

Comment:
Why is `not` singled out ? As far as I can tell, the SMTLIB specification does not treat `not` specifically (from a syntax point of view) and I'd like to keep it that way if possible (i.e. keep the lexer/parser as close to the formal syntax definition as possible).

Reply:
It is fine to me. This shall be dealt at the lecture of the theory (e.g., 'not' is in the core theory) for the type checker.